packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -428,6 +428,12 @@ export class MergeTree {
      */
     public pendingSegments: List<SegmentGroup> | undefined;
     private segmentsToScour: Heap<LRUSegment> | undefined;
+    /**
+     * Whether or not all blocks in the mergeTree currently have information about local partial lengths computed.
+     * This information is only necessary on reconnect, and otherwise costly to bookkeep.
+     * This field enables tracking whether partials need to be recomputed using localSeq information.
+     */
+    private localPartialsComputed = false;
```

Comment:
still going through this, but wondering if this could be encapsulated in the partial length structure somehow, rather than needing the merge tree to track this

Reply:
Yeah, I considered a few options before settling on this one. I think to encapsulate the state in PartialSequenceLengths, you'd need `PartialSequenceLengths` objects to do something like retain the reference to the block that created them so they can choose to recompute on demand. I think it's a bit more than I want to take on in this PR though, and potentially worse perf (memory impact of only have one field per tree that tracks this vs one field per node does feel nice).

I think if we decide to refactor this, we might also reconsider the overall structure of how mergeTree creates and updates its partial lengths on each node.